src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -230,12 +230,17 @@ def create_filesystem(self, fs_name, metadata_pool, data_pool):
                    'data': data_pool}
         return self.mgr.mon_command(command)
 
-    def remove_filesystem(self, fs_name):
+    def remove_filesystem(self, fs_name, confirm):
         command = {'prefix': 'fs fail', 'fs_name': fs_name}
         r, outb, outs = self.mgr.mon_command(command)
         if r != 0:
             return r, outb, outs
-        command = {'prefix': 'fs rm', 'fs_name': fs_name, 'yes_i_really_mean_it': True}
+
+        if confirm is not None and confirm == "--yes-i-really-mean-it":
+            command = {'prefix': 'fs rm', 'fs_name': fs_name, 'yes_i_really_mean_it': True}
+        else:
+            command = {'prefix': 'fs rm', 'fs_name': fs_name, 'yes_i_really_mean_it': False}
```

Comment:
Is it correct to check the --yes-i-really-mean-it flag after the previous 'fs fail' mon command? It seems like you might want to check for confirmation before making any changes.

Reply:
Yes, I think that's better. Do you mean before the `command = {'prefix': 'fs fail', 'fs_name': fs_name}` line?

Reply:
Yes, exactly. That means you'll want to do a separate check for it and bail out early instead of just passing the right value to "fs rm". In fact, once you have that you may want to just unconditionally call "fs rm" with 'yes_i_really_mean_it' : True.

Reply:
@jtlayton Fixed now.